File: go/word-count/word_count.go
package wc

import (
	"regexp"
	"strings"
)

func WordCount(text string) Histogram {
	hist := Histogram{}

	wordMatcher := regexp.MustCompile(`\w+`)

	words := wordMatcher.FindAll([]byte(strings.ToLower(text)), -1)

	for _, word := range words {
		hist[string(word)]++
	}

	return hist
}

File: scala/word-count/src/main/scala/word_count.scala
class Phrase(phrase: String) {
    def wordCount() = {
        words.map {word => (word, countWord(word))}.toMap
    }

    def words() = {
        "[\\w']+".r.findAllIn(phrase.toLowerCase).toList
    }

    def countWord(word: String) = {
        words.count(_==word)
    }
}